Removes Slick profile defautls

rcavalcanti opened the pull request.

rcavalcanti commented on README.md:
```diff
@@ -42,7 +42,7 @@ Configure `akka-persistence`:
 
 Configure `slick`:
-- The following slick drivers are supported:
+- The following slick profiles are supported:
```
this and subsequent references are point to `profiles` not `drivers`

rcavalcanti commented on src/main/resources/reference.conf:
```diff
@@ -46,7 +46,8 @@ jdbc-journal {
   slick {
-    driver = "slick.jdbc.PostgresProfile$"
+    profile = "slick.jdbc.PostgresProfile$"
+    profile = ${?jdbc-journal.slick.driver}
```
This will make sure that an user that have declared the `driver`, but not the `profile` will have his driver config used as a `profile`

rcavalcanti commented on src/main/resources/reference.conf:
```diff
@@ -103,7 +104,8 @@ jdbc-snapshot-store {
   slick {
-    driver = "slick.jdbc.PostgresProfile$"
+    profile = "slick.jdbc.PostgresProfile$"
+    profile = ${?jdbc-snapshot-store.slick.driver}
```
same as above

rcavalcanti commented on src/main/resources/reference.conf:
```diff
@@ -191,7 +193,8 @@ jdbc-read-journal {
   slick {
-    driver = "slick.jdbc.PostgresProfile$"
+    profile = "slick.jdbc.PostgresProfile$"
+    profile = ${?jdbc-read-journal.slick.driver}
```
same as above

WellingR commented:
Hey @rcavalcanti thanks for your contribution. I have actually already included a part of these changes already. However your addition of `profile = ${?jdbc-journal.slick.driver}` is a good one.

Could you rebase your branch against the latest master and resolve the conflict? Then I will merge this PR

rcavalcanti commented:
@WellingR rebased and conflict resolved

rcavalcanti commented on src/main/resources/reference.conf:
```diff
@@ -47,6 +47,7 @@ jdbc-journal {
 
   slick {
     profile = "slick.jdbc.PostgresProfile$"
+    profile = ${?jdbc-journal.slick.driver}
```
I did some local test to simulate the case in which an user have configured `driver` and we kind of move it to `profile`. 

That will only work if `driver = slick.jdbc.PostgresProfile$`.

If the user has it as `driver = slick.driver.PostgresDriver$`, we are in trouble. Because Slick will see it as the value of `profile` property and try to load `slick.driver.PostgresDriver`. And that will fail because there is not such a class anymore, only a type alias.

WellingR replied:
I see your point. In that case it would actually be better to not include this setting.

In the end it might actually make sense to remove all references to postgress at all. In this case a misconfigured database configuration would fail to load (hopefully with a useful error message). Removing this default would actually prevent issues such as #116

rcavalcanti replied:
It's true that the `reference.conf` should not contain any settings that are specific to environment. I think that it's often used as a reference for documentation, but that's not the best thing to do because we end up setting default values that can be very confusing.

rcavalcanti commented:
@WellingR I pushed a workaround for it. A ugly one. 
Not yet sure we must take that path, but let's discuss it next week. I need a fresh mind on Monday to evaluate it. 

Eventually, this is something we should try to solve in Slick itself.

We are facing this same issue in Lagom. The reason for that is because here and there (Lagom) we are providing default drivers in a `reference.conf` file. That's great for quick bootstrap, but now that we want / need to promote `slick.profile` over `slick.driver` we run the risk of overriding properties in production system.

WellingR commented:
A potential solution would be to comment the lines specifying the profile, driver (and perhaps also the other environment related settings such as url, user, password etc).

These commented lines serve as documentation in the reference conf file. But since we cannot provide sensible defaults, it it actually better to not provide these defaults at all.

I ran some of the tests where I removed the profile, which gives a good error message. The same holds if the url is not specified. Somehow if the jdbc driver is not specified the tests seem to pass, maybe slick determines this based on the url.

@rcavalcanti what do you think about this?

rcavalcanti commented:
@WellingR I also think this is the best to do. I will adapt this PR accordingly and remove the workaround I added. 

For the test, it may be that the are overriding those values which would explain why it still works. In any case, the test should define their own values and production code must not define defaults.

rcavalcanti commented:
@WellingR I updated this PR. 

- removed workaround added ealier
- commented out the database specific settings in `reference.conf`
- removed references to boot2docker and docker (doesn't make sense to have it there)
- minor tweaks in comments in `reference.conf`

rcavalcanti commented:
BTW, this PR can be easily cherrypicked into akka-2.4 branch
